tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+been changed by us ("changed/not changed by user"), and whether the default
+content has been changed by the package maintainer ("same as/changed from
+default"):
+
+| The conffile is...          | **same as default**  | **changed from default**  |
+| ---                         | ---                  | ---                       |
+| **...changed by user**      | Keep user's changes  | Ask user                  |
+| **...not changed by user**  | No changes to make   | Apply changes from update |
+
+### Identifying conffiles
+
+Out of the list of files in a package, how do we know which ones are the
+conffiles?
+
+After all, they are not marked by any particular file extension, and
+although they are often found in the `/etc/` directory, they don't *have* to be
+there. As we saw before, the only thing conffiles have in common is that the
+package maintainer decided to mark them as such.
+
+But that's our clue! So once more, `dpkg` can come to our rescue. The
+following command will show us the subset of files in a package that have been
+marked as "conffiles" by the maintainer:
+
+```bash
+dpkg-query --show -f='${Conffiles}\n' apache2
+```
+
+Unlike `dpkg --listfiles`, this query *also* gives us a string of letters and
+numbers. This string is known as the **"MD5 checksum"** or **"MD5 hash"**. 
+
+```text
+ /etc/apache2/apache2.conf 354c9e6d2b88a0a3e0548f853840674c
+ /etc/apache2/conf-available/charset.conf e6fbb8adf631932851d6cc522c1e48d7
+ /etc/apache2/conf-available/security.conf 332668933023a463046fa90d9b057193
+ /etc/apache2/envvars e4431a53c868ae0dfcde68564f3ce6a7
+ /etc/apache2/magic a6d370833a02f53db6a0a30800704994
+[...]
+```
+
+We see the checksum of a specific file by running this command:
+
+```bash
+md5sum /etc/apache2/apache2.conf
+```
+
+Which returns us the checksum followed by the file:
+
+```bash
+354c9e6d2b88a0a3e0548f853840674c  /etc/apache2/apache2.conf
+```
+
+You might well be wondering "why do we care about that?" since they match (in
+this example).
+
+The checksum is like a fingerprint - it's unique for every *version* of a file,
+so any time the file is changed it will get a new checksum -- which allows us
+to see **if a file has been changed** from the default. 
+
+### Verifying checksums
+
+Let's set up a situation so we can poke a bit at this idea. We can start by
+making some changes to a conffile. In Apache2, the main conffile is
+`/etc/apache2/apache2.conf`, so let's use that. In a situation where we are
+setting up a new webserver, we might reasonably want to increase the `LogLevel`
+from "warn" to "debug" to get more debugging messages, so let's run this
+command and use `sed` to make that change in the conffile:
+
+```bash
+sudo sed -e 's/LogLevel warn/LogLevel debug/' -i /etc/apache2/apache2.conf
+```
+
+Then, we'll restart our Apache2 server so that we can activate our configuration
+changes:
+
+```bash
+sudo systemctl restart apache2
```

Comment:
After running this, I got `Failed to restart apache2.service: Unit apache2.service not found.`

I don't think the tutorial actually has us reinstall apache2, so that should be added (unless I just missed it)

Reply:
Do you know, you're completely right. In all the previous steps I forgot to add that in! Great catch

Reply:
Fixed